### PR TITLE
Auth, injection, deduping, whitelisting added

### PR DESCRIPTION
**Pull request by tohst**
Fixed an issue with the first PR

### CODE REVIEW

File: inject.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    for i, bulk in enumerate(bulks):
+        if i % 100 == 0:
+            print(f'Done {i} records!')
+        try:
+            single_bulk_to_es(bulk, config, config['with_retry'], es)
+        except Exception as e:
+            log('warn', 'Chunk {i} got exception ({e}) while processing'.format(e=e, i=i))
+
+
+def format_msg(msg, sevirity):
```
Comment (gordongrace) [inline inject.py]: Is this supposed to be `severity` ?

File: whitelist.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    'ca_x-funnelback-title-for-duplicate-detection','ca_x-funnelback-undesirable-text',
+)
+dta_wl = (
+    'DOCURL','HOST','PORTFOLIO','REPORT_ID','3LD','m_x-dta-agds','m_x-dta-agds-count','m_x-dta-uikit',
+    'm_x-dta-uikit-count','m_x-dta-captcha','m_x-dta-captcha-count','m_x-dta-twitter',
+    'm_x-dta-twitter-count','m_x-dta-facebook','m_x-dta-facebook-count','m_x-dta-canonical',
+    'm_x-dta-canonical-count','m_x-dta-ga-id','m_x-dta-ga-360-id','m_x-dta-reported-lang',
+    'm_x-dta-detected-lang','m_x-funnelback-twitter-hash-tags','m_x-funnelback-twitter-user-tags',
+    'm_x-dta-links-clickhere','m_x-dta-links-clickhere-value','m_x-dta-links-clickhere-count',
+    'm_x-dta-h1-count','m_x-dta-title-length','m_x-dta-title-length-size','m_x-dta-weasel-words',
```
Comment (gordongrace) [inline whitelist.py:197]: I'd love to replace 'weasel-words' with 'non-preferred-terms'

Comment (tohst) [reply whitelist.py:197]: This script works with those fields as already defined multiple steps in the pipleine prior.

I think this is more appropriate on a ticket about updating the data dictionary/re-filtering

**Review by gordongrace — COMMENTED** [review]
Getting us close - there appears to be a typo and non-preferred term ('weasel words') in this version, but otherwise good to go.

File: inject.py (hunk 1 above)
Comment (tohst) [reply inject.py]: Fixed. Actually went with 'level' as log levels are technically a mixture of priority/severity/hierarchy depending on context.

File: whitelist.py (hunk 2 above)
Comment (tohst) [reply whitelist.py:197]: Future proofish compromise: I added 'm_x-dta-non-preferred-terms' and 'm_x-dta-non-preferred-terms-count' so once the data dictionary and subsequently the post-crawl filter scripts are updated to use those names instead it should already work.

**Review by gordongrace — COMMENTED** [review]
All comments resolved.

**Pull request merged**